js/go112_wasm.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 The GopherWasm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build wasm
+// +build go1.12
+
+package js
+
+import (
+	"syscall/js"
+)
+
+type Func = js.Func
+
+func FuncOf(fn func(this Value, args []Value) interface{}) Func {
+	return js.FuncOf(fn)
+}
+
+// Callback is for backward compatibility. Use Func instead.
+type Callback = js.Func
+
+// EventCallbackFlag is for backward compatibility.
+type EventCallbackFlag int
+
+const (
+	PreventDefault EventCallbackFlag = 1 << iota
+	StopPropagation
+	StopImmediatePropagation
+)
+
+// NewCallback is for backward compatibility. Use FuncOf instead.
+func NewCallback(fn func([]Value)) Callback {
+	return FuncOf(func(this Value, args []Value) interface{} {
+		go func() {
+			fn(args)
+		}()
+		return nil
+	})
+}
+
+// NewEventCallback is for backward compatibility. Use FuncOf instead.
+func NewEventCallback(flags EventCallbackFlag, fn func(event Value)) Callback {
```

Comment:
This seems like code duplication with the GopherJS case. Can't you use the same wrapper around `FuncOf` for both Go 1.12 and GopherJS?

Reply:
Hmm, I don't think it is easy to unify since `js.Value` should be emulated by `*js.Object` on GopherJS side, so the codes are a little different.

Reply:
If `FuncOf` already unifies wasm and GopherJS, then there should be no specific code in `NewEventCallback` any more.

Reply:
Ah, that makes sense.

> If FuncOf already unifies wasm and GopherJS, 

You mean that if `FuncOf` is defined on both Wasm and GopherJS, `NewCallback` and `NewEventCallback` implementation should be common, right?

Reply:
Done.